First version

ferag opened the pull request.

Commit:
First version of authenticate

ferag committed:
First version to configure N IdPs

alvarolopez commented:
Can you please rebase this over the branch master so that CI takes care of running the tests? The following should do the work:

    git checkout master
    git pull origin master
    git checkout first_version
    git rebase master
    git push origin first_version

Commit:
testing: solving problems

Solved style issues and fixed unit testing

ferag committed:
Removing duplicated code not needed

alvarolopez committed:
Set up environment so that federation code can do its job

We should not mimic downstream code, but rather set up the required ENV
vars to ensure that the assertions can be consumed downstream by all the
federation plumbing.

alvarolopez committed:
Add better management configuration

Rather than relying on dictionaries we can rely on different sections,
prefixed by openid, like:

    [openid_foo]
    issuer = foo
    client_id = foobar

    [openid_bar]
    issuer = bar
    client_id = foobarbaz

alvarolopez committed:
tox: update unit tests configuration

alvarolopez committed:
Generate configuration for our module

Add a sample configuration file

alvarolopez committed:
requirements: pin pyoidc version

alvarolopez committed:
Refactor authenticate method

alvarolopez committed:
Improve option handling and register all options that we are using

Keystone federation registers the "remote_id_attribute" option for each
of the protocols. We can register it ourselves, so that it gets
documented in the generated configuration file.

alvarolopez committed:
Update metadata and add a README with some documentation
